pkg/virtctl/vnc/screenshot/screenshot.go

```diff
@@ -0,0 +1,76 @@
+package screenshot
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+var fileName string
+
+func NewScreenshotCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "screenshot (VMI)",
+		Short:   "Create a VNC screenshot of a virtual machine instance.",
+		Example: usage(),
+		Args:    templates.ExactArgs("screenshot", 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := Screenshot{clientConfig: clientConfig}
+			return c.Run(cmd, args)
+		},
+	}
+	cmd.Flags().StringVarP(&fileName, "file", "f", "", "where to store the VNC screenshot in PNG format. User '-' for stdout")
+	err := cmd.MarkFlagRequired("file")
+	if err != nil {
+		panic(err)
+	}
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	return cmd
+}
+
+func usage() string {
+	return `   # Take a VNC screenshot of 'testvmi' in png format:
+   {{ProgramName}} vnc screenshot testvmi -f screenshot.png
+
+   # Take a VNC screenshot of 'testvmi' in png format and pipe it through "display" to show it right away:
+   {{ProgramName}} vnc screenshot testvmi -f - | display`
+}
+
+type Screenshot struct {
+	clientConfig clientcmd.ClientConfig
+}
+
+func (o *Screenshot) Run(_ *cobra.Command, args []string) error {
+	namespace, _, err := o.clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+
+	vmi := args[0]
+
+	virtCli, err := kubecli.GetKubevirtClientFromClientConfig(o.clientConfig)
+	if err != nil {
+		return err
+	}
+
+	// setup connection with VM
+	screenshot, err := virtCli.VirtualMachineInstance(namespace).Screenshot(vmi)
+	if err != nil {
+		return fmt.Errorf("Can't access VMI %s: %v: %s", vmi, err, string(screenshot))
```

Comment:
I think if you just convert `[]byte` as `string(screenshot)` it will print garbage and potentially unprintable or special characters. I would personally omit this part, but alternatively it can be encoded to e.g. base64

Reply:
The main issue is that the error is very unspecific and the body contains the actual error (a general problem in our webhooks I think). Maybe I can find a solution to add it to the actual error.

Reply:
So maybe I misunderstood it, but I got the impression that `screenshot` will contain only the PNG bits, no?
 Or do you mean that in the error case, it will include the rest response?

Reply:
yes, if there is a server error the err.Error() string just contains .e.g `500 internal error`, and the actual reason for the error is in the body. That does not look right, due to potential issues, like you describe, but it is in virtctl at this moment the only way to print the actual reason for the error.

Reply:
And depending on when it fails, it could indeed add some png parts already, resulting to ugly output. I will try to see if I can fix this the right way.